### ISSUE TITLE
[HOLD for payment 2024-03-25] [$500] Can't unselect category when editing a split bill

### ISSUE THREAD

**Issue by m-natarajan** [Expensify/App#37146]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 1.4.43-14
**Reproducible in staging?:** y
**Reproducible in production?:** y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Expensify/Expensify Issue URL:**
**Issue reported by:** @DylanDylann 
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1708412951140219

## Action Performed:
1. Go to workspace expense chat
2. Click + button beside to composer
3. Click Split Bill
4. Click Scan tab
5. Choose a receipt (any image)
6. Go to confirmation page
7. Enter a category
8. Click Split
9. Unselect category
## Expected Result: 
Category should be removed
## Actual Result: 
Category still remain

## Workaround:
unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [x] Android: Native
- [x] Android: mWeb Chrome
- [x] iOS: Native
- [x] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [x] MacOS: Desktop

## Screenshots/Videos

Add any screenshot/video evidence

https://github.com/Expensify/App/assets/38435837/3304d031-be25-41bf-b480-ad47966b69c5

https://github.com/Expensify/App/assets/38435837/45cff15d-c0dc-4f00-a1ef-242f2662d509

</details>

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~018323df92a9ee320a</li>
        <li>Upwork Job ID: 1761062942437457920</li>
        <li>Last Price Increase: 2024-03-08</li>
    </ul>
</details>

**Comment by DylanDylann** [comment]
Hi everyone, I saw this bug while reviewing [this PR](https://github.com/Expensify/App/pull/35458). I agree that this bug happened before and It isn't caused by [this PR](https://github.com/Expensify/App/pull/35458). But I am not sure whether it's an intentional or a bug.

**Comment by DylanDylann** [comment]
I have more context [here](https://github.com/Expensify/App/pull/35458/files#r1494420048), If we need to speed up this issue, I could take over this one

**Comment by wildan-m** [comment]
@DylanDylann how can we make the category mandatory? I've asked it [here](https://expensify.slack.com/archives/C01GTK53T8Q/p1708742487863599), but haven't received an answer yet.

What I've tried: Created a control workspace in oldot and enabled the must-use category, but with no success.

**Comment by wildan-m** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.

Unable to deselect options while editing a split bill. The options may include category or tag.

### What is the root cause of that problem?

For the category issue, we are currently utilizing `category.searchText` instead of `updatedCategory` for updating the category.

https://github.com/Expensify/App/blob/5eed6322bd2e559c684481e8bc0122b0825779e7/src/pages/iou/request/step/IOURequestStepCategory.js#L94-L98

For the tag, the problem lies in our use of `transaction` instead of `splitDraftTransaction`.

https://github.com/Expensify/App/blob/5eed6322bd2e559c684481e8bc0122b0825779e7/src/pages/iou/request/step/IOURequestStepTag.js#L70-L71

### What changes do you think we should make in order to solve the problem?

We can revise this line:
https://github.com/Expensify/App/blob/5eed6322bd2e559c684481e8bc0122b0825779e7/src/pages/iou/request/step/IOURequestStepCategory.js#L95

to
```
            IOU.setDraftSplitTransaction(transaction.transactionID, {category: updatedCategory});
```

And modify the transaction here:

https://github.com/Expensify/App/blob/5eed6322bd2e559c684481e8bc0122b0825779e7/src/pages/iou/request/step/IOURequestStepTag.js#L70-L71

to
```
    const transactionToUse = isEditing && isSplitBill ? splitDraftTransaction : transaction;
    const transactionTag = TransactionUtils.getTag(transactionToUse);
    const tag = TransactionUtils.getTag(transactionToUse, tagIndex);    
```

Where `splitDraftTransaction` is

```
        splitDraftTransaction: {
            key: ({route}) => {
                const transactionID = lodashGet(route, 'params.transactionID', 0);
                return `${ONYXKEYS.COLLECTION.SPLIT_TRANSACTION_DRAFT}${transactionID}`;
            },
        },
```

[Branch to test.](https://github.com/Expensify/App/compare/main...wildan-m:App:wildan/fix/37146-cant-unselect-options?expand=1)
### What alternative solutions did you explore? (Optional)

**Comment by puneetlath** [comment]
@eVoloshchak thoughts on the proposals?

**Comment by wildan-m** [comment]
@dukenv0307 Thanks for your concern. I can't reproduce [this category regression](https://github.com/Expensify/App/issues/37346) with tag, so provide empty check for tag might be optional.

If the OP knows that this similar unselect issue occured for tag as well, I believe He/She will include it in the bug description. For me it does make sense to solve similar issue in one PR, similar to what we have done [here](https://github.com/Expensify/App/pull/26690).

**Comment by eVoloshchak** [comment]
> For the tag, the problem lies in our use of transaction instead of splitDraftTransaction.

@wildan-m, the behavior is different than for category, this does look like a separate bug, I think we should handle it in a separate issue

I think we should proceed with @dukenv0307's [proposal](https://github.com/Expensify/App/issues/37146#issuecomment-1961621022) 

🎀👀🎀 C+ reviewed!

**Comment by puneetlath** [comment]
Sorry for the delay! Proposal sounds good to me.

**Comment by puneetlath** [comment]
@eVoloshchak friendly reminder about the checklist so that we can pay on Monday.

**Comment by puneetlath** [comment]
@dukenv0307 offer is here: https://www.upwork.com/nx/wm/offer/101566868

@eVoloshchak still waiting on the checklist.

**Comment by eVoloshchak** [comment]
- The PR that introduced the bug has been identified. Link to the PR: this bug stems from the initial implementation of https://github.com/Expensify/App/pull/28618, it was partially resolved by https://github.com/Expensify/App/pull/38205, but this specific case (scan split bill) was missed. Judging by the method name (`resetMoneyRequestCategory_temporaryForRefactor`) it was planned to fix this in the subsequent PR(s), but that wasn't the case
- The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake. Link to comment: N/A
- A discussion in #expensify-bugs has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner. Link to discussion: I don't think an additional discussion is needed, this is a simple bug that was missed during testing, I think a regression test is what's necessary here

**Comment by eVoloshchak** [comment]
**Regression Test Proposal**

1. Go to workspace expense chat
2. Click + button beside the composer
3. Click Split Bill
4. Click Scan tab
5. Choose a receipt (any image)
6. Go to confirmation page
7. Enter a category
8. Click Split
9. Click on the created IOU request -> Category
10. Unselect the category
11. Verify that category is removed

**Do we agree 👍 or 👎**

**Comment by puneetlath** [comment]
>(resetMoneyRequestCategory_temporaryForRefactor) it was planned to fix this in the subsequent PR(s), but that wasn't the case

>The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake. Link to comment: N/A


Can we comment on the PR in that case? It's not ideal to create a temporary method and then never finish it. Especially since it led to a bug.

**Comment by puneetlath** [comment]
@eVoloshchak bump!

**Comment by eVoloshchak** [comment]
Apologies, I was wrong about the PR that has caused this. 
As discussed in https://github.com/Expensify/App/pull/35458/files#r1494844943, this is the old logic carried over from `EditSplitBillPage`. Updated the [checklist](https://github.com/Expensify/App/issues/37146#issuecomment-2018211506)

**Comment by puneetlath** [comment]
Payment summary: 
- C - $500 - @dukenv0307 - paid on Upwork
- C+ - $500 - @eVoloshchak - to be paid on NewDot

@eVoloshchak please request on NewDot. Thanks everyone!

**Comment by JmillsExpensify** [comment]
$500 approved for @eVoloshchak

**Comment by puneetlath** [comment]
Weird, you're right. I went ahead and paid it.
